File: app/src/main/java/com/example/weatherapp/CityLoader.java
package com.example.weatherapp;

import android.content.Context;
import androidx.loader.content.AsyncTaskLoader;
import java.util.ArrayList;

public class CityLoader extends AsyncTaskLoader<City> {
    String cityQuery;
    String tempType;

    public CityLoader(Context context, String cityQuery, String tempType) {
            super(context);
            this.cityQuery = cityQuery;
            this.tempType = tempType;
        }

    @Override
    protected void onStartLoading() {
            forceLoad();
        }

    @Override
    public City loadInBackground() {
        if (cityQuery == null) return null;
        return DataQuery.fetchCityData(cityQuery, tempType);
    }
}

File: app/src/main/java/com/example/weatherapp/CityLoader.kt
package com.example.weatherapp

import android.content.Context
import androidx.loader.content.AsyncTaskLoader
import com.example.weatherapp.DataQuery.fetchCityData

class CityLoader(context: Context?, var cityQuery: String?) : AsyncTaskLoader<City?>(context!!) {
    override fun onStartLoading() {
        forceLoad()
    }

    override fun loadInBackground(): City? {
        return if (cityQuery == null) null else fetchCityData(cityQuery!!)
    }
}